### PR TITLE
Configuration to delete parsed files

### LINKED ISSUE #259: Don't write parsed files property
**Problem**
Streams are duplicately stored in the database and on the filesystem. The streams on the filesystem are not used by the code after downloaded, parsed and inserted into the database. With the sheer volume of data, these files could eat up a lot of disk space.

**Solution**
A configuration property to delete files after successfully processing them instead of moving them to the parsed directory. I think it should be turned on by default for us, but left as an option in case others want to reconstruct the hashgraph from the raw data. Additionally, we should rename the parser persist options to avoid confusion.

- Add property: `hedera.mirror.parser.balance.keepFiles=false`
- Add property: `hedera.mirror.parser.event.keepFiles=false`
- Add property: `hedera.mirror.parser.record.keepFiles=false`
- Rename balance parsed directory from `parsedBalanceFiles` to `parsed`
- Rename event parsed directory from `parsedEventStreamFiles` to `parsed`
- Rename record parsed directory from `parsedRecordFiles` to `parsed`
- Move `hedera.mirror.parser.record.persistClaims` -> `hedera.mirror.parser.record.persist.claims`
- Move `hedera.mirror.parser.record.persistContracts` -> `hedera.mirror.parser.record.persist.contracts`
- Move `hedera.mirror.parser.record.persistCryptoTransferAmounts` -> `hedera.mirror.parser.record.persist.cryptoTransferAmounts`
- Move `hedera.mirror.parser.record.persistFiles` -> `hedera.mirror.parser.record.persist.files`
- Move `hedera.mirror.parser.record.persistNonFeeTransfers` -> `hedera.mirror.parser.record.persist.nonFeeTransfers`
- Move `hedera.mirror.parser.record.persistSystemFiles` -> `hedera.mirror.parser.record.persist.systemFiles`
- Move `hedera.mirror.parser.record.persistTransactionBytes` -> `hedera.mirror.parser.record.persist.transactionBytes`
- Update documentation

**Alternatives**
Add more disk space, manually delete, script to delete

**Additional Context**

### PR DESCRIPTION
**Pull request by apeksharma**
**Detailed description**:
Signed-off-by: Apekshit Sharma <apekshit.sharma@hedera.com>

**Which issue(s) this PR fixes**:
partially fixes #259 

**Special notes for your reviewer**:

**Checklist**
- [x] Documentation added
- [x] Tests updated

### CODE REVIEW

**Comment by apeksharma** [comment]
This PR is not addressing other suggestions in #259. I starting doing them, but seemed that they'll become point of trivial back-and-forth, and not worth the time cost of review iterations.
It feels best that anyone feeling particular about dir/configs in those suggestions does it directly.

**Comment by steven-sheehy** [comment]
So now we will have trivial back and forth about why did you take a task if you weren't going to do all the sub-tasks?

**Comment by apeksharma** [comment]
well i meant to when i took it. But during work, realized that it's actually two tasks:
\- first one, remove parsed files, important 
\- second one, config/directory renames, not urgent
Seeing no reason to be done together, I did the part that's actually needed now.
The main thing is out of the way, and when the other part is done i'll quickly review and approve it. I hope it makes sense and this path seems okay.

File: hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/balance/BalanceFileParser.java [hunk 1]
```diff
@@ -97,23 +94,26 @@ private void processAllFilesForHistory() {
...
-                // move it
-                Utility.moveFileToParsedDir(balanceFile.getCanonicalPath(), "/parsedBalanceFiles/");
+                moveOrDeleteParsedFile(balanceFile.getCanonicalPath());
             }
         } catch (Exception e) {
             log.error("Error processing balances files after {}", stopwatch, e);
         }
     }
+
+    private void moveOrDeleteParsedFile(String fileName) {
```
Comment (Nana-EC) [inline hedera-mirror-importer/src/main/java/com/hedera/mirror/importer/parser/balance/BalanceFileParser.java]: Should this method be something that lives in Utility class. Taking in a boolean of whether to move or delete and the name
e.g. 
```
public void moveOrDeleteParsedFile(boolean move, String fileName) {
        if (moved) {
             Utility.moveFileToParsedDir(fileName, "/parsedBalanceFiles/");
         } else {
             Utility.deleteParsedFile(fileName);
         }
}
```

That way The FileParser classes only have to call 
- `Utility. moveOrDeleteParsedFile(parserProperties.isKeepFiles(), PARSED_DIR)`

File: docs/configuration.md [hunk 2]
```diff
@@ -114,10 +114,13 @@ value, it is recommended to only populate overridden properties in the custom `a
 | `hedera.mirror.parser.balance.batchSize`                   | 2000                    | The number of balances to insert before committing                                             |
 | `hedera.mirror.parser.balance.enabled`                     | true                    | Whether to enable balance file parsing                                                         |
 | `hedera.mirror.parser.balance.fileBufferSize`              | 200000                  | The size of the buffer to use when reading in the balance file                                 |
+| `hedera.mirror.parser.balance.keepFiles`                   | false                   | Whether to keep parsed files after successful parsing. If false, files are deleted.            |
```
Comment (Nana-EC) [inline docs/configuration.md]: The code defaults to true but this file says it's false. You should clarify which one is right. I think we're going for false by default now no?

**Pull request merged**